### PR TITLE
restore auto-derivation for show

### PR DESCRIPTION
**Pull request by kailuowang**
@joroKr21 mind to take a look? 
Once this is approved I'll add more. 
I also namespaced the implicit methods to avoid potential conflicts.

### CODE REVIEW

File: core/src/main/scala/cats/derived/show.scala [hunk 1]
```diff
@@ -21,23 +21,16 @@ object MkShow extends MkShowDerivation {
...
-
-trait MkShow0 extends MkShow1 {
+trait MkShowDerivation extends MkShow1 {
   implicit val emptyProductDerivedShow: MkShow[HNil] =
     instance(_ => "")
 
   implicit def productDerivedShowFurther[K <: Symbol, V, T <: HList](
        implicit key: Witness.Aux[K],
-       showV: Lazy[MkShow[V]],
+       showV: Show[V] OrElse MkShow[V],
```
Comment (joroKr21) [inline core/src/main/scala/cats/derived/show.scala]: I think we should do this for coproducts as well. Although less likely that someone would define a special case only for one subclass, it's still a possibility.

Comment (kailuowang) [reply core/src/main/scala/cats/derived/show.scala]: thanks I missed that.

**Comment by milessabin** [comment]
Looks good, but there are a couple of renamings that appear to be unrelated to the essence of the PR: `mkEmpty` -> `kittensMkEmpty` and `mkFunctor` -> `kittensMkFunctor`. Why are they included here?

**Comment by kailuowang** [comment]
@milessabin they are to avoid implicit conflicts. In this PR, I introduced a new one of the same kind, namely `kittenMkShow`. To make it consistent I renamed the other two and hopefully setup a convention for naming schema. We can certainly do that in a separate PR if you prefer.

**Comment by milessabin** [comment]
What do they conflict with?

**Comment by kailuowang** [comment]
Nothing in particular at the moment, but if some other lib for some reason also has an implicit method called `mkShow` which isn't very inconceivable, it will conflict with our import. This is to avoid the potential implicit name conflict when both libs are imported.

**Comment by milessabin** [comment]
Gotcha. OK, no objections ... LGTM :-)

**Comment by kailuowang** [comment]
thanks!

**Pull request merged**